# Importing Libraries 

In [2]:
import pandas as pd
import seaborn as sns 
import matplotlib.pyplot as plt
import numpy as np 


### 1. Introduction

- The objective of this analysis is to explore the Iris dataset to identify patterns, relationships, and key features that distinguish different species. The analysis focuses on understanding data distribution, correlations, and feature importance to support data-driven insights.

### 2. Tools & Dataset
- Tools: Python (Pandas, Matplotlib, Seaborn)
- Dataset: Iris Dataset

### 3. Loading and Prearing Dataset

In [4]:
from sklearn.datasets import load_iris
import pandas as pd

iris = load_iris()
df = pd.DataFrame(iris.data, columns=iris.feature_names)
df['species'] = iris.target

# Convert labels
df['species'] = df['species'].map({
    0: 'setosa',
    1: 'versicolor',
    2: 'virginica'
})

df.head()

,sepal length (cm),sepal width (cm),petal length (cm),petal width (cm),species
0,5.1,3.5,1.4,0.2,setosa
1,4.9,3.0,1.4,0.2,setosa
2,4.7,3.2,1.3,0.2,setosa
3,4.6,3.1,1.5,0.2,setosa
4,5.0,3.6,1.4,0.2,setosa


### 4. Data Overview

In [5]:
df.info()
df.describe()

<class 'pandas.DataFrame'>
RangeIndex: 150 entries, 0 to 149
Data columns (total 5 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   sepal length (cm)  150 non-null    float64
 1   sepal width (cm)   150 non-null    float64
 2   petal length (cm)  150 non-null    float64
 3   petal width (cm)   150 non-null    float64
 4   species            150 non-null    str    
dtypes: float64(4), str(1)
memory usage: 6.0 KB


,sepal length (cm),sepal width (cm),petal length (cm),petal width (cm)
count,150.000000,150.000000,150.000000,150.000000
mean,5.843333,3.057333,3.758000,1.199333
std,0.828066,0.435866,1.765298,0.762238
min,4.300000,2.000000,1.000000,0.100000
25%,5.100000,2.800000,1.600000,0.300000
50%,5.800000,3.000000,4.350000,1.300000
75%,6.400000,3.300000,5.100000,1.800000
max,7.900000,4.400000,6.900000,2.500000


##### Insight:
- Dataset contains 150 rows and 5 columns
- Features are numerical and suitable for analysis

### 5. Summary Statistics 

In [8]:
df.groupby('species').mean()

,sepal length (cm),sepal width (cm),petal length (cm),petal width (cm)
species,,,,
setosa,5.006,3.428,1.462,0.246
versicolor,5.936,2.770,4.260,1.326
virginica,6.588,2.974,5.552,2.026


##### Insight:
- Setosa has smaller petal measurements
- Virginica has the largest measurements

In [2]:
# Summary Statistics
df.info()
print('\n')
print(df.describe())

NameError: name 'df' is not defined

In [ ]:
# Species Distribution
print(df['species'].value_counts())

In [ ]:
# Correlation Heatmap
corr = df.drop('species', axis=1).corr()
plt.figure(figsize=(8,6))
sns.heatmap(corr, annot=True, cmap='coolwarm')
plt.title('Correlation Heatmap')
plt.show()

In [ ]:
# Pairplot
sns.pairplot(df, hue='species')
plt.show()

In [ ]:
# Boxplots
plt.figure(figsize=(10,6))
for i, col in enumerate(df.columns[:-1]):
    plt.subplot(2,2,i+1)
    sns.boxplot(x='species', y=col, data=df)
    plt.title(col)
plt.tight_layout()
plt.show()

In [ ]:
# Histograms
plt.figure(figsize=(10,6))
for i, col in enumerate(df.columns[:-1]):
    plt.subplot(2,2,i+1)
    sns.histplot(df[col], kde=True)
    plt.title(col)
plt.tight_layout()
plt.show()

# EDA Report

## Summary
The Iris dataset consists of 150 samples of iris flowers from three species: Setosa, Versicolor, and Virginica. Each sample has four features: sepal length, sepal width, petal length, and petal width.

## Key Findings
- **Dataset Balance**: The dataset is perfectly balanced with 50 samples for each species.
- **Correlations**: 
  - Petal length and petal width show the highest correlation (0.96).
  - Sepal length correlates moderately with petal length (0.87) and petal width (0.82).
  - Sepal width has low correlation with other features.
- **Species Patterns**:
  - Setosa species has distinctly smaller petal dimensions compared to Versicolor and Virginica.
  - Versicolor and Virginica show some overlap in sepal measurements but are more separable by petal features.
  - Setosa can be easily distinguished from the other two species based on petal length (< 2 cm) and petal width (< 0.8 cm).
- **Distributions**: Most features show roughly normal distributions, with some skewness in petal measurements.

## Visual Insights
- The pairplot reveals clear clusters for Setosa and partial separation for the other species.
- Boxplots confirm Setosa's distinct characteristics and highlight outliers in Versicolor and Virginica.
- Histograms show the spread of measurements across species.

This EDA provides a foundation for further analysis, such as classification modeling.